src/node_http2.cc

```diff
@@ -2646,7 +2646,7 @@ void Http2Session::New(const FunctionCallbackInfo<Value>& args) {
       static_cast<SessionType>(
           args[0]->Int32Value(env->context()).ToChecked());
   Http2Session* session = new Http2Session(state, args.This(), type);
-  session->get_async_id();  // avoid compiler warning
+  USE(session->get_async_id());  // avoid compiler warning
```

Comment:
The get_async_id() call is unnecessary here.
```suggestion
  USE(session);  // avoid compiler warning
```
Does this work?

Reply:
In fact, the session variable is passed to the Debug() function in the next line, which means that it is being used already. So we should be able to remove this line too, right?
```suggestion
```

Reply:
@mhdawson the line was added in https://github.com/nodejs/node/commit/69e6c5a212622ec15b8c2cf904480b6582c6c3a5 when we used a different macro - [`DEBUG_HTTP2SESSION`](https://github.com/nodejs/node/blame/bd85844c4e80c7aa1fc02a986c7619c3956b0061/src/node_http2.cc#L2207) for this. Later on the macro was changed to the Debug() function we see now in https://github.com/nodejs/node/commit/15c7a49bfca1db23f4f7a3a79aea04a0075fbd62, so I don't think it is necessary anymore.

Reply:
The reason why this ran into a compiler warning before was because of https://github.com/nodejs/node/commit/69e6c5a212622ec15b8c2cf904480b6582c6c3a5#diff-430b8a0d2f4cc14e0eac16d13f3879eed2697ebe3749acbb48a11db567a7e936R67:

(oversimplification of how the macro was defined)
```cc
#ifdef NODE_DEBUG_HTTP2
#define DEBUG_HTTP2(...) debug_vfprintf(__VA_ARGS__);
 #define DEBUG_HTTP2SESSION(session, message)                                  \
   do {                                                                        \
     DEBUG_HTTP2("Http2Session %s (%.0lf) " message "\n",                      \
                    session->TypeName(),                                       \
                    session->get_async_id());                                  \
   } while (0)
#else
#define DEBUG_HTTP2SESSION(...) do {} while (0)
#endif
```

so it was essentially a no-op if NODE_DEBUG_HTTP2 was not defined.

Reply:
> @RaisinTen, My assumption was that it was there because the Debug() function does not always use the session.

FWIW, USE() too doesn't do anything with the passed variable: https://github.com/nodejs/node/blob/3c4ee5267a522a8f16907c4b3726dd4afb29d4f6/src/util.h#L520-L522

Reply:
@RaisinTen thanks for the detailed explanation, removed.